### PR TITLE
Add docs for HelpText to CheckboxGroup and RadioGroup

### LINKED ISSUE #3439: Description and error message props for useCheckbox, useCheckboxGroup, and useRadioGroup
# 🙋 Feature Request

Provide `descriptionProps` and `errorMessageProps` for `useCheckbox`, `useCheckboxGroup`, and `useRadioGroup`. This would make them consisten with every other field in react-aria.

## 🤔 Expected Behavior

This should behave exactly the same as in the `useTextField` implementation.

## 😯 Current Behavior

No support for `descriptionProps` or `errorMessageProps` is currently provided for checkboxes or radio groups. 

## 💁 Possible Solution

Implement in a similar manner to `useTextField`.

## 🔦 Context

Not providing description and error message props makes it more difficult to provide a consistent experience when implementing error messaging in forms.

## 💻 Examples

`const { groupProps, labelProps, descriptionProps, errorMessageProps } = useCheckboxGroup(props, state);`

### PR DESCRIPTION
**Pull request by reidbarber**
Closes #3439

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx [hunk 1]
```diff
@@ -196,6 +198,31 @@ function Example() {
 }
 ```
 
+### Description
```
Comment (LFDanLu) [inline packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx:201]: Doesn't look like the description/error message for these checkbox group/radio group aria docs are being properly associated with the group itself (no aria-describedby).

File: packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx [hunk 2]
```diff
@@ -196,6 +198,31 @@ function Example() {
...
+</CheckboxGroup>
+```
+
+### Error message
+
+The `errorMessage` prop can be used to help the user fix a validation error. It should be combined with the `validationState` prop to
+semantically mark the checkbox group as invalid for assistive technologies.
+
+```tsx example
+<CheckboxGroup label="Favorite sports" errorMessage="Invalid selection of sports.">
```
Comment (LFDanLu) [inline packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx:219]: should you provide `validationState` here?

File: packages/@react-aria/radio/docs/useRadioGroup.mdx [hunk 3]
```diff
@@ -274,6 +278,29 @@ function Example() {
...
+</RadioGroup>
+```
+
+### Error message
+
+The `errorMessage` prop can be used to help the user fix a validation error. It should be combined with the `validationState` prop to
+semantically mark the radio group as invalid for assistive technologies.
+
+```tsx example
+<RadioGroup label="Favorite pet" errorMessage="Invalid pet selection.">
```
Comment (LFDanLu) [inline packages/@react-aria/radio/docs/useRadioGroup.mdx:298]: Same here, needs validationState?

**Review by LFDanLu — CHANGES_REQUESTED** [review]
Also seeing some weirdness where the spectrum components aren't getting the right aria-describedby either

File: packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx (hunk 2 above)
Comment (reidbarber) [reply packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx]: I think that's better, just updated. We aren't doing this for other examples, but maybe we should update those as well.

File: packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx (hunk 1 above)
Comment (ktabors) [reply packages/@react-aria/checkbox/docs/useCheckboxGroup.mdx:203]: I don't see this when I look at the docs for TextField #help text.

File: packages/@react-spectrum/radio/docs/RadioGroup.mdx [hunk 4]
```diff
@@ -185,6 +185,41 @@ and "end" refers to the right most edge. For right-to-left (RTL) languages, this
...
+  return (
+    <RadioGroup
+      aria-label="Favorite pet"
+      onChange={setSelected}
+      validationState={isValid ? 'valid' : 'invalid'}
+      description="Please select a pet."
+      errorMessage={
+        selected === 'cats'
+          ? 'No cats allowed.'
+          : 'Please select dogs.'
```
Comment (ktabors) [inline packages/@react-spectrum/radio/docs/RadioGroup.mdx:207]: Can you change the message to "No cats or dogs allows"?

File: packages/@react-spectrum/checkbox/docs/CheckboxGroup.mdx [hunk 5]
```diff
@@ -228,6 +228,36 @@ and "end" refers to the right most edge. For right-to-left (RTL) languages, this
...
+    <CheckboxGroup
+      label="Pets"
+      onChange={setChecked}
+      value={checked}
+      validationState={isValid ? 'valid' : 'invalid'}
+      description="Select only dogs and dragons."
+      errorMessage={
+        checked.includes('cats')
+          ? 'No cats allowed.'
+          : 'Select only dogs and dragons.'
```
Comment (ktabors) [inline packages/@react-spectrum/checkbox/docs/CheckboxGroup.mdx:251]: It's only valid when both dogs and cats are selected, so if I have one selected I get the non cats allowed message even when I haven't selected cats.

**Review by ktabors — COMMENTED** [review]
Curious about the aria-describedby since that doesn't seem to be there for other components that have this.

File: packages/@react-spectrum/radio/docs/RadioGroup.mdx (hunk 4 above)
Comment (snowystinger) [reply packages/@react-spectrum/radio/docs/RadioGroup.mdx:207]: but dogs are allowed?

File: packages/@react-spectrum/checkbox/docs/CheckboxGroup.mdx (hunk 5 above)
Comment (snowystinger) [reply packages/@react-spectrum/checkbox/docs/CheckboxGroup.mdx:251]: but it's valid when you have dogs and dragons selected? not cats

Comment (lydiashortridge36) [reply packages/@react-spectrum/checkbox/docs/CheckboxGroup.mdx:251]: Amazing

File: packages/@react-spectrum/radio/docs/RadioGroup.mdx (hunk 4 above)
Comment (lydiashortridge36) [reply packages/@react-spectrum/radio/docs/RadioGroup.mdx:207]: Dogs

Comment (lydiashortridge36) [reply packages/@react-spectrum/radio/docs/RadioGroup.mdx:207]: > Can you change the message to "No cats or dogs allows"?

[]() dogs allows

File: packages/@react-spectrum/checkbox/src/CheckboxGroup.tsx [hunk 6]
```diff
@@ -93,6 +93,15 @@ function CheckboxGroup(props: SpectrumCheckboxGroupProps, ref: DOMRef<HTMLDivEle
...
             {children}
+            {hasHelpText &&
+              <HelpText
+                descriptionProps={descriptionProps}
+                errorMessageProps={errorMessageProps}
+                description={description}
+                errorMessage={errorMessage}
+                validationState={validationState}
+                showErrorIcon={showErrorIcon} />
+            }
```
[the remark this answers is not in the document]
Comment (reidbarber) [reply packages/@react-spectrum/checkbox/src/CheckboxGroup.tsx:104]: Missed this in the merge, thanks for catching

**Review by snowystinger — APPROVED** [review]
lgtm

**Review by LFDanLu — APPROVED** [review]
LGTM

**Pull request merged**